test/unit/storage_subformat_vagrant_libvirt_test.py

```diff
@@ -1,87 +1,61 @@
-from mock import call, patch
-import mock
-
-from .test_helper import patch_open
-from .storage_subformat_vagrant_base_test import TestDiskFormatVagrant
+from textwrap import dedent
+from mock import (
+    patch, Mock
+)
 
 from kiwi.storage.subformat.vagrant_libvirt import DiskFormatVagrantLibVirt
 
-from textwrap import dedent
 
-
-class TestDiskFormatVagrantLibVirt(TestDiskFormatVagrant):
+class TestDiskFormatVagrantLibVirt(object):
     def setup(self):
-        super(TestDiskFormatVagrantLibVirt, self).setup()
+        xml_data = Mock()
+        xml_data.get_name = Mock(
+            return_value='some-disk-image'
+        )
+        self.xml_state = Mock()
+        self.xml_state.xml_data = xml_data
+        self.xml_state.get_image_version = Mock(
+            return_value='1.2.3'
+        )
+        self.vagrantconfig = Mock()
+        self.vagrantconfig.get_virtualsize = Mock(
+            return_value=42
+        )
         self.disk_format = DiskFormatVagrantLibVirt(
             self.xml_state, 'root_dir', 'target_dir',
             {'vagrantconfig': self.vagrantconfig}
         )
-
-    def test_store_to_result(self):
-        result = mock.Mock()
-        self.disk_format.store_to_result(result)
-        result.add.assert_called_once_with(
-            compress=False,
-            filename='target_dir/'
-            'some-disk-image.x86_64-1.2.3.vagrant.libvirt.box',
-            key='disk_format_image',
-            shasum=True,
-            use_for_bundle=True
-        )
+        assert self.disk_format.image_format == 'vagrant.libvirt.box'
+        assert self.disk_format.provider == 'libvirt'
 
     @patch('kiwi.storage.subformat.vagrant_libvirt.Command.run')
-    @patch('kiwi.storage.subformat.vagrant_base.mkdtemp')
     @patch('kiwi.storage.subformat.vagrant_libvirt.DiskFormatQcow2')
-    @patch('kiwi.storage.subformat.vagrant_base.random.randrange')
-    @patch_open
-    def test_create_image_format(
-        self, mock_open, mock_rand, mock_qcow, mock_mkdtemp, mock_command
+    def test_create_box_img(
+        self, mock_qcow, mock_command
     ):
-        mock_rand.return_value = 0xa
-        mock_mkdtemp.return_value = 'tmpdir'
-        qcow = mock.Mock()
+        qcow = Mock()
         qcow.image_format = 'qcow2'
         mock_qcow.return_value = qcow
-        context_manager_mock = mock.Mock()
-        mock_open.return_value = context_manager_mock
-        file_mock = mock.Mock()
-        enter_mock = mock.Mock()
-        exit_mock = mock.Mock()
-        enter_mock.return_value = file_mock
-        setattr(context_manager_mock, '__enter__', enter_mock)
-        setattr(context_manager_mock, '__exit__', exit_mock)
-        metadata_json = dedent('''
-            {
-              "format": "qcow2",
-              "provider": "libvirt",
-              "virtual_size": "42"
-            }
-        ''').strip()
-        vagrantfile = dedent('''
```

Comment:
Could we leave the check for the whole Vagrantfile in? When we only test `get_additional_vagrant_config_settings`, it would be possible that the template still creates a malformed Vagrantfile (I know, that is unlikely since we have tests for the template in place).

Reply:
yes let's check the merging. I'll extend the base class test

Reply:
done